File: ui/src/lib/help.ts
export type HelpContext =
  | "general"
  | "editor"
  | "journal"
  | "graph"
  | "flashcards"
  | "tasks"
  | "chat"
  | "settings"
  | "sync"
  | "search";

const HELP_PAGES: Record<HelpContext, string> = {
  general: "Help - Grafium Guide",
  editor: "Help - Editor",
  journal: "Help - Journal Guide",
  graph: "Help - Graph",
  flashcards: "Help - Flashcards",
  tasks: "Help - Tasks",
  chat: "Help - Chat",
  settings: "Help - Settings",
  sync: "Help - Sync",
  search: "Help - Search",
};

export function isHelpContext(value: string | null): value is HelpContext {
  return value !== null && value in HELP_PAGES;
}

export function helpPageTitle(context: HelpContext): string {
  return HELP_PAGES[context];
}

export async function loadHelpPage(context: HelpContext): Promise<string> {
  const { invoke } = await import("@tauri-apps/api/core");
  return invoke<string>("help_get_page", { context });
}

File: ui/src-tauri/src/commands/help.rs
const HELP_PAGES: &[(&str, &str)] = &[
    (
        "general",
        include_str!("../../resources/welcome/pages/Help - Grafium Guide.md"),
    ),
    (
        "editor",
        include_str!("../../resources/welcome/pages/Help - Editor.md"),
    ),
    (
        "journal",
        include_str!("../../resources/welcome/pages/Help - Journal Guide.md"),
    ),
    (
        "graph",
        include_str!("../../resources/welcome/pages/Help - Graph.md"),
    ),
    (
        "flashcards",
        include_str!("../../resources/welcome/pages/Help - Flashcards.md"),
    ),
    (
        "tasks",
        include_str!("../../resources/welcome/pages/Help - Tasks.md"),
    ),
    (
        "chat",
        include_str!("../../resources/welcome/pages/Help - Chat.md"),
    ),
    (
        "settings",
        include_str!("../../resources/welcome/pages/Help - Settings.md"),
    ),
    (
        "sync",
        include_str!("../../resources/welcome/pages/Help - Sync.md"),
    ),
    (
        "search",
        include_str!("../../resources/welcome/pages/Help - Search.md"),
    ),
];

#[tauri::command(rename_all = "camelCase")]
pub fn help_get_page(context: String) -> Result<String, String> {
    HELP_PAGES
        .iter()
        .find(|(name, _)| *name == context)
        .map(|(_, content)| (*content).to_string())
        .ok_or_else(|| format!("Unknown help context: {context}"))
}
